### DISCUSSION TITLE
SQL Syntax error

### DISCUSSION THREAD

**Discussion by MathiasWeisheit** [spatie/laravel-stats#26] [General]
```
<?php
namespace App\Models\Stats;
use Spatie\Stats\BaseStats;
class SearchStartStats extends BaseStats {}
```
`
$stats = SearchStartStats::query()->get();`

Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '(PARTITION BY yearweek(created_at, 3) ORDER BY `id` DESC) AS rn, `stats_events`.' at line 1 (SQL: select ROW_NUMBER() OVER (PARTITION BY yearweek(created_at, 3) ORDER BY `id` DESC) AS rn, `stats_events`.*, yearweek(created_at, 3) as period from `stats_events` where (`name` = SearchStartStats) and `type` = set and `created_at` >= 2022-05-23 08:21:26 and `created_at` < 2022-06-23 08:21:26)",

**Comment by freekmurze** [comment]
According to our tests, the package should work. Maybe you're using another flavour or version of MySQL version that us. We'd accept a PR that fixes it for you if the code is not too complicated.

**Reply by MathiasWeisheit** [reply]
Hi, thanks for the fast response! Great work @ spatie!
Im using 5.7.38-1

**Reply by freekmurze** [reply]
Maybe try MySQL 8 👍

**Reply by MathiasWeisheit** [reply]
Sorry my fault!

**Reply by MathiasWeisheit** [reply]
Do you have an fallback for my version?

**Reply by freekmurze** [reply]
Nope...

**Comment by MathiasWeisheit** [comment]
can be closed.
Thx
